Source/JavaScriptCore/jit/AssemblyHelpers.cpp

```diff
@@ -820,12 +820,13 @@ void AssemblyHelpers::emitAllocateWithNonNullAllocator(GPRReg resultGPR, const J
 #endif
 
 #if CPU(ARM64)
-    // On ARM64, we can leverage instructions like load-pair and shifted-add to make loading from the free list 
+    // On ARM64, we can leverage instructions like load-pair and shifted-add to make loading from the free list
     // and extracting interval information use less instructions.
 
     // Assert that we can use loadPairPtr for the interval bounds and nextInterval/secret.
-    RELEASE_ASSERT(FreeList::offsetOfIntervalEnd() - FreeList::offsetOfIntervalStart() == sizeof(uintptr_t)); 
-    RELEASE_ASSERT(FreeList::offsetOfSecret() - FreeList::offsetOfNextInterval() == sizeof(uintptr_t)); 
+    RELEASE_ASSERT(FreeList::offsetOfIntervalEnd() - FreeList::offsetOfIntervalStart() == sizeof(uintptr_t));
+    RELEASE_ASSERT(FreeList::offsetOfNextInterval() - FreeList::offsetOfIntervalEnd() == sizeof(uintptr_t));
+    RELEASE_ASSERT(FreeList::offsetOfSecret() - FreeList::offsetOfNextInterval() == sizeof(uintptr_t));
```

Comment:
Is it possible to make these `static_assert`s?

Or if it has to be `RELEASE_ASSERT`s, can put these in a constructor somewhere so that we only pay this price once since these are static constraints?  Not essential to do in this patch.

Reply:
I'm pretty sure this can't be `static_assert` because it uses `reinterpret_cast` to get the offset. I'd also guess that the compiler will remove the ASSERT though since it probably proves it can't trigger anyway.

Reply:
I think we should just change it to ASSERT. This path is incredibly frequently executed, so ASSERT is enough to catch in Debug tests.

Reply:
And offsetOf etc. does not work with static_assert as Keith said.